internal/envoy/versions.go

```diff
@@ -20,30 +20,121 @@ import (
 	"fmt"
 	"io"
 	"net/http"
+	"strconv"
+	"strings"
 
+	"github.com/tetratelabs/func-e/internal/globals"
 	"github.com/tetratelabs/func-e/internal/version"
 )
 
-// FuncEVersions returns a version map from a remote URL. eg globals.DefaultEnvoyVersionsURL.
-func FuncEVersions(ctx context.Context, envoyVersionsURL string, p version.Platform, v version.Version) (version.ReleaseVersions, error) {
+type funcEVersionsGetter func() (version.ReleaseVersions, error)
+
+// FuncEVersions fetches a version map form a remote URL and gets the latest patched version given
+// a version input. This input can be major.minor.patch (e.g. 1.1.1) or major.minor (1.1).
+type FuncEVersions interface {
+	// Get returns a version map from a remote URL. e.g. from globals.DefaultEnvoyVersionsURL.
+	Get() (version.ReleaseVersions, error)
+
+	// FindLatestVersions returns the latest patch version given an Envoy version.
+	FindLatestVersions(version.Version) (string, error)
```

Comment:
speaking aloud as going to be offline 🌴 one neat thing about this being an interface is that code to conditionally call FindLatestPatch is easy to guard in the cmd package.

Ex. you could have code that accepts this interface and the test cases will be easy to write as the impl can raise an error on Get() and do whatever you are testing on FindLatestVersions both without needing to worry about http calls and not requiring to bring most version parsing internal here.